ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -143,13 +143,23 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
         }
     }
 
-    public async showNodePicker(expectedContextValue: string): Promise<IAzureNode> {
-        const picks: PickWithData<SubscriptionNode>[] = this._subscriptionNodes.map((n: SubscriptionNode) => new PickWithData<SubscriptionNode>(n, n.treeItem.label, n.subscription.subscriptionId));
-        let node: AzureNode = (await this._ui.showQuickPick<SubscriptionNode>(picks, localize('selectSubscription', 'Select a Subscription'))).data;
+    public async showNodePicker(expectedContextValues: string | string[], startingNode?: IAzureNode): Promise<IAzureNode> {
```

Comment:
Curious here, since I saw two choices of going about this : 
1. Accept a string or string array and do the conversion of string to string array inside the function. 
2. Ask all call sites to pass in a string array. 
I personally think approach two would be clearer to the users of this function.

Reply:
The biggest reason for number 2 was so that it would be backwards compatible (which is the best practice when you're publishing an npm package like this - and I just didn't want to go back and change every call of `showNodePicker` to an array)

We could debate which one is clearer (I happen to think 2 is clearer since most of the time you will pass in one value), but like I said that wasn't the main reason I did it this way